projects/vdk-plugins/vdk-jobs-troubleshooting/src/vdk/plugin/jobs_troubleshoot/api/troubleshoot_utility.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+from abc import abstractmethod
+
+
+class ITroubleshootUtility:
+    """
+    Allows for troubleshoot utilities to be implemented. It provides a unified
+    API to be used by the jobs troubleshooting plugin irrespective of the
+    utility's underlying logic.
+    """
+
+    @abstractmethod
+    def start(self):
+        """
+        Starts the execution of the utility when the job is initialized.
+
+        NOTE: Any troubleshooting utility needs to be self-contained and when
```

Comment:
What do you mean ? It's a good comment .

But also the framework should take of that. Basically both start and stop would ignore any error/exception coming from them and only print them.

Reply:
Agree, the framework needs to take care of this. Did expand the doc string to cover it.